easyapplybot.py

```diff
@@ -251,31 +282,45 @@ def get_easy_apply_button(self):
 		return EasyApplyButton
 
 
+	def is_jsonable(self, x):
+		try:
+			json.dumps(x)
+			return True
+		except:
+			return False
+
 	def send_resume(self):
 		def is_present(button_locator):
-			return len(self.browser.find_elements(button_locator[0],
-													 button_locator[1])) > 0
+			return (len(self.browser.find_elements(button_locator[0], button_locator[1])) > 0)
 
 		try:
-			time.sleep(random.uniform(1.5, 2.5))
-			next_locater = (By.CSS_SELECTOR,
-							"button[aria-label='Continue to next step']")
-			review_locater = (By.CSS_SELECTOR,
-							"button[aria-label='Review your application']")
-			submit_locater = (By.CSS_SELECTOR,
-								"button[aria-label='Submit application']")
-			submit_application_locator = (By.CSS_SELECTOR,
-									"button[aria-label='Submit application']")
-			error_locator = (By.CSS_SELECTOR,
-							"p[data-test-form-element-error-message='true']")
-			upload_locator = (By.CSS_SELECTOR, "input[name='file']")
+
+			time.sleep(3)
+			log.info("Attempting to send resume")
+			#TODO These locators are not future proof. These labels could easily change. Ideally we would search for contained text;
+			# was unable to get it to work using XPATH and searching for contained text
+			upload_locator = (By.CSS_SELECTOR, "label[aria-label='DOC, DOCX, PDF formats only (2 MB).']")
+			next_locator = (By.CSS_SELECTOR, "button[aria-label='Continue to next step']")
+			review_locator = (By.CSS_SELECTOR, "button[aria-label='Review your application']")
+			submit_locator = (By.CSS_SELECTOR, "button[aria-label='Submit application']")
+			submit_application_locator = (By.CSS_SELECTOR, "button[aria-label='Submit application']")
+			error_locator = (By.CSS_SELECTOR, "p[data-test-form-element-error-message='true']")
+			cover_letter = (By.CSS_SELECTOR, "input[name='file']")
```

Comment:
```suggestion
			upload_locator = (By.CSS_SELECTOR, "input[name='file']")
```

Reply:
Keeping the upload_locator as _any_ input path lets us handle cases like Photo, Cover Letter, and Resume from the configuration file.

Reply:
I have seen the window have multiple uploads on the same screen.
I dont think using the same locator for multiple cases on the same window/screen will work, right?

Reply:
I am confused to if you are suggesting that we need to upload both a resume and cover letter? I have never seen an option to upload a cover letter.

Should we delete one of them?

I think the order of operations within the filling out the application screens is becoming a bit confusing.
The master branch has the cover locator pulled out of the for loop and set to happen before any of the other locators. This assumes that the first thing to do is upload a resume, but for me I always see the next button first and then the user can try to upload a document second. 

I think what we need to do is create a state machine for the application pop up screen(s) so that way we can follow what is happening easier and update quicker in the future if something changes or new issues come up.

Reply:
So the current workflow inside the `send_resume` function in master:

![image](https://user-images.githubusercontent.com/6612207/88403901-2275ec80-cd9b-11ea-9528-0824fe75aab1.png)